third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -24,6 +25,73 @@ static bool isLocalLoadOrDotLayoutConversion(Operation *op) {
   return false;
 }
 
+// Copy of mlir::getBackwardSlice with changes to handle nested regions.
+// This is a temporary local fix until these changes are upstreamed to mlir.
+static void getDeepBackwardSlice(Operation *op,
```

Comment:
Consider making this fix to the MLIR code base, perhaps through `BackwardSliceOptions`?

Reply:
Yeah we should consider landing such general changes to upstream mlir. Then we can update llvm submodule to bring it in.